WWSTCERT-11943 Added support for Bosch Motion Detector II [+M]

JanJakubiszyn opened the pull request:
Check all that apply

# Type of Change

- [ ] WWST Certification Request
     - If this is your first time contributing code:
          - [x ] I have reviewed the README.md file
          - [ x] I have reviewed the CODE_OF_CONDUCT.md file
          - [ x] I have signed the CLA
     - [ ] I plan on entering a WWST Certification Request or have entered a request through the WWST Certification console at developer.smartthings.com
- [ ] Bug fix
- [ ] New feature
- [ x] Refactor

# Checklist

- [ x] I have performed a self-review of my code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have verified my changes by testing with a device or have communicated a plan for testing
- [ ] I am adding new behavior, such as adding a sub-driver, and have added and run new unit tests to cover the new behavior

# Description of Change


# Summary of Completed Tests

JanJakubiszyn committed:
Added support for Bosch matter motion sensor

# Conflicts:
#	drivers/SmartThings/matter-switch/src/switch_utils/device_configuration.lua
#	drivers/SmartThings/matter-switch/src/switch_utils/fields.lua

hcarter-775 commented on drivers/SmartThings/matter-switch/profiles/light-level-motion-illuminance-battery-temp.yml:
```diff
@@ -0,0 +1,29 @@
+name: light-level-motion-illuminance-battery-temp
```
Would you mind changing the name and file to `light-level-battery-illuminance-motion-temperature` to more closely fit the naming convention already created. Thanks!

JanJakubiszyn committed:
Implementing changes from the code review: edit of the device profile name and capability order, edit of the match_profile's condition

# Conflicts:
#	drivers/SmartThings/matter-switch/src/switch_utils/device_configuration.lua

# Conflicts:
#	drivers/SmartThings/matter-switch/src/switch_utils/device_configuration.lua

greens commented:
@JanJakubiszyn You'll need to rebase your changes.

hcarter-775 reviewed (commented):
other than these last edge-case nits, this lgtm. Thanks for the prompt responses!

JanJakubiszyn committed:
Code cleanup

KKlimczukS committed:
Update fingerprints.yml

KKlimczukS committed:
Merge branch 'main' into Bosch_motion_sensor_support

JanJakubiszyn committed:
Merge branch 'main' into Bosch_motion_sensor_support

KKlimczukS committed:
Merge branch 'main' into Bosch_motion_sensor_support

hcarter-775 commented on drivers/SmartThings/matter-switch/profiles/light-level-battery-illuminance-motion-temperature.yml:
```diff
@@ -0,0 +1,29 @@
+        version: 1
+      - id: illuminanceMeasurement
+        version: 1
+      - id: switch
+        version: 1
+      - id: switchLevel
+        version: 1
+        config:
+          values:
+            - key: "level.value"
+              range: [1, 100]
```
To keep profile consistency, these should be the first two listed capabilities.

The third should be one that is currently unlisted, i.e.
```
  - id: statelessSwitchLevelStep
    version: 1
```

KKlimczukS replied:
@hcarter-775 device profile re-arranged

hcarter-775 commented on drivers/SmartThings/matter-switch/profiles/light-level-battery-illuminance-motion-temperature.yml:
```diff
@@ -0,0 +1,29 @@
+      - name: MotionSensor
+preferences:
+  - preferenceId: tempOffset
+    explicit: true
```
add newline

KKlimczukS replied:
done

KKlimczukS committed:
Update light-level-battery-illuminance-motion-temperature.yml

hcarter-775 reviewed (approved):
LGTM
